### PR TITLE
cargo fuzz harness

### PR DESCRIPTION
**Pull request by evanrichter**
simple [cargo-fuzz](https://github.com/rust-fuzz/cargo-fuzz) based harness

run with `cargo +nightly fuzz run filter`

### CODE REVIEW

**Review by sejoker — APPROVED** [review]

File: fuzz/corpus/filter/test.xml [hunk 1]
```diff
@@ -0,0 +1 @@
+../tests/test.xml
```
Comment (xjewer) [inline fuzz/corpus/filter/test.xml]: symlinks are incorrect, should be:

```
test.xml -> ../../../tests/test.xml
```

File: fuzz/corpus/filter/filtered.xml [hunk 2]
```diff
@@ -0,0 +1 @@
+../tests/filtered.xml
```
Comment (xjewer) [inline fuzz/corpus/filter/filtered.xml]: symlinks are incorrect, should be:

```
filtered.xml -> ../../../tests/filtered.xml
```

**Review by xjewer — CHANGES_REQUESTED** [review]

File: fuzz/corpus/filter/test.xml (hunk 1 above)
Comment (evanrichter) [reply fuzz/corpus/filter/test.xml]: oops! fixed now

File: fuzz/corpus/filter/filtered.xml (hunk 2 above)
Comment (evanrichter) [reply fuzz/corpus/filter/filtered.xml]: fixed now

File: fuzz/corpus/filter/test.xml (hunk 1 above)
Comment (xjewer) [reply fuzz/corpus/filter/test.xml]: thank you

**Comment by xjewer** [comment]
fuzzer helped to find this issue https://github.com/netvl/xml-rs/issues/210

**Review by xjewer — APPROVED** [review]

**Pull request merged**